frontends/concrete-python/concrete/fhe/mlir/converter.py

```diff
@@ -618,6 +618,20 @@ def tlu(self, ctx: Context, node: Node, preds: List[Conversion]) -> Conversion:
         variable_input_index = variable_input_indices[0]
         variable_input = preds[variable_input_index]
 
+        if variable_input.origin.properties.get("name") == "truncate_bit_pattern":
+            original_bit_width = variable_input.origin.properties["original_bit_width"]
+            lsbs_to_remove = variable_input.origin.properties["kwargs"]["lsbs_to_remove"]
+            truncated_bit_width = original_bit_width - lsbs_to_remove
+
+            if variable_input.bit_width > original_bit_width:
```

Comment:
shouldn't this be done in the general case, as discussed on slack ?

Reply:
No, you saw the figures and the values on the documentation right? That's the appropriate execution semantics, actual bit width reduction is an optimization for TLU.

Reply:
If you want to extract the msbs directly, you can use right shift (which we'll optimize to use something similar soon), or you can use bit extraction (which we'll introduce soon).